src/python/WMCore/Services/DBS/DBS3Reader.py

```diff
@@ -352,7 +354,9 @@ def crossCheckMissing(self, datasetPath, *lfns):
         are *not* known by DBS
 
         """
-        allLfns = self.dbs.listFileArray(dataset=datasetPath, validFileOnly=1, detail=False)
+        allLfns = []
+        for fileDict in self.dbs.listFileArray(dataset=datasetPath, validFileOnly=1, detail=False):
```

Comment:
same comment here, it will be better to put `self.dbs.listFileArray` into `try/except` block